### ISSUE TITLE
Old version of GitHub Runners was deprecated as soon as a new version is released

### ISSUE THREAD

**Issue by zepeng811** [actions/runner#3405]
**Describe the bug**
we are getting runner problems this morning indicating version `v2.317.0` is deprecated, but the latest version `v2.318.0` was only released today.
```
[RUNNER 2024-07-26 17:07:23Z ERR  Terminal] WRITE ERROR: An error occured: Error: Forbidden Runner version v2.317.0 is deprecated and cannot receive messages.
An error occured: Error: Forbidden Runner version v2.317.0 is deprecated and cannot receive messages.
```

the [doc](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/autoscaling-with-self-hosted-runners#controlling-runner-software-updates-on-self-hosted-runners:~:text=If%20you%20do%20not%20perform%20a%20software%20update%20within%2030%20days%2C%20the%20GitHub%20Actions%20service%20will%20not%20queue%20jobs%20to%20your%20runner.%20In%20addition%2C%20if%20a%20critical%20security%20update%20is%20required%2C%20the%20GitHub%20Actions%20service%20will%20not%20queue%20jobs%20to%20your%20runner%20until%20it%20has%20been%20updated.) indicated we should have 30 days to update before GHA stop queue jobs for the runners.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run workflows with runner version `2.317.0`
2. notices the runner are terminated immediately with the above error

**Expected behaviour**
Old versions of runners should still be able to queue jobs until the 30 days mark to prevent breaking change 

## Runner Version and Platform
Version of your runner?
v2.317.0

OS of the machine running the runner? OSX/Windows/Linux/...
Linux amd64

## What's not working?
Please include error messages and screenshots.
runner version is deprecated and cannot receive messages

## Job Log Output
```
[RUNNER 2024-07-26 17:07:23Z ERR  Terminal] WRITE ERROR: An error occured: Error: Forbidden Runner version v2.317.0 is deprecated and cannot receive messages.
An error occured: Error: Forbidden Runner version v2.317.0 is deprecated and cannot receive messages.
[RUNNER 2024-07-26 17:07:23Z ERR  Listener] GitHub.DistributedTask.WebApi.AccessDeniedException: Error: Forbidden Runner version v2.317.0 is deprecated and cannot receive messages.
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Actions.RunService.WebApi.BrokerHttpClient.GetRunnerMessageAsync(Nullable`1 sessionId, String runnerVersion, Nullable`1 status, String os, String architecture, Nullable`1 disableUpdate, CancellationToken cancellationToken)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Common.BrokerServer.<>c__DisplayClass7_0.<<GetRunnerMessageAsync>b__0>d.MoveNext()
[RUNNER 2024-07-26 17:07:23Z ERR  Listener] --- End of stack trace from previous location ---
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Common.RunnerService.RetryRequest[T](Func`1 func, CancellationToken cancellationToken, Int32 maxRetryAttemptsCount, Func`2 shouldRetry)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.MessageListener.GetNextMessageAsync(CancellationToken token)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.Runner.RunAsync(RunnerSettings settings, Boolean runOnce)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.Runner.RunAsync(RunnerSettings settings, Boolean runOnce)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.Runner.RunAsync(RunnerSettings settings, Boolean runOnce)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.Runner.ExecuteCommand(CommandSettings command)
[RUNNER 2024-07-26 17:07:23Z ERR  Listener]    at GitHub.Runner.Listener.Program.MainAsync(IHostContext context, String[] args)
Runner listener exit with terminated error, stop the service, no retry needed.
Exiting runner...
```

**Comment by tomleicircle** [comment]
Hey GitHub folks, with all the outages from GitHub Actions nowadays, this is just plainly unacceptable, can we escalate with GitHub management on the reliability issue?

**Comment by luketomlinson** [comment]
hey @zepeng811 this should be resolved now. Restarting the runner should get you back up and running.

**Comment by vinaysandela** [comment]
We are on RHEL 7.9 and all our runners hosted on it were offline. Tried restarting and configuring a new one, but no luck. Can you help. 

```
sudo ./svc.sh status

/etc/systemd/system/actions.runner.enterprises-Org.AR26.service
● actions.runner.enterprises-Org.AR26.service - GitHub Actions Runner (enterprises-Org.AR26)
   Loaded: loaded (/etc/systemd/system/actions.runner.enterprises-Org.AR26.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Wed 2024-07-31 21:03:51 CDT; 2min 19s ago
  Process: 35128 ExecStart=/opt/tfs/isolated-GitHubARs/AR26/runsvc.sh (code=exited, status=0/SUCCESS)
 Main PID: 35128 (code=exited, status=0/SUCCESS)

Jul 31 21:03:50 Server systemd[1]: Started GitHub Actions Runner (enterprises-Org.AR26).
Jul 31 21:03:50 Server runsvc.sh[35128]: .path=/home/SA/.asdf/shims:/home/SA/.asdf/bin:/opt/maven//bin:/opt/rh/rh-dotnet60/root/usr...psvcdevops
Jul 31 21:03:51 Serverrunsvc.sh[35128]: Starting Runner listener with startup type: service
Jul 31 21:03:51 Serverrunsvc.sh[35128]: Started listener process, pid: 35165
Jul 31 21:03:51 Server runsvc.sh[35128]: Started running service
Jul 31 21:03:51 Serverrunsvc.sh[35128]: /opt/tfs/isolated-GitHubARs/AR26/bin/Runner.Listener: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found ....Listener)
Jul 31 21:03:51 Server runsvc.sh[35128]: /opt/tfs/isolated-GitHubARs/AR26/bin/Runner.Listener: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found ....Listener)
Jul 31 21:03:51 Server runsvc.sh[35128]: Runner listener exited with error code 1
Jul 31 21:03:51 Serverrunsvc.sh[35128]: Runner listener exit with terminated error, stop the service, no retry needed.
Hint: Some lines were ellipsized, use -l to show in full.
```

**Comment by TingluoHuang** [comment]
@vinaysandela a recent released (2.318.0) runner consumed latest dotnet version which deprecates support on Redhat 7 since it's end of support based on [Redhat](https://access.redhat.com/product-life-cycles?product=Red%20Hat%20Enterprise%20Linux,OpenShift%20Container%20Platform%204)

We also noticed this issue with Redhat 7 and have paused the new runner rollout.

You can keep use 2.317.0 runner which works fine on Redhat 7.

Do you have any plan to upgrade to Redhat 8 in the near future?

The runner is currently on dotnet 6 which will end its LTS later this year, this means we are going to be forced to dotnet 8 which doesn't support Redhat 7. 😢
